lib/src/footer.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Parsing footer lines from commit messages.
+
+use indexmap::IndexMap;
+
+/// Parse the footer lines from a commit message; these are simple key-value
+/// pairs, separated by a colon, describing extra information in a commit
+/// message; an example is the following:
+///
+/// ```text
+/// chore: fix bug 1234
+///
+/// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+/// tempor incididunt ut labore et dolore magna aliqua.
+///
+/// Co-authored-by: Alice <alice@example.com>
+/// Co-authored-by: Bob <bob@example.com>
+/// Reviewed-by: Charlie <charlie@example.com>
+/// Change-Id: I1234567890abcdef1234567890abcdef12345678
+/// ```
+///
+/// In this case, there are four footer lines: two `Co-authored-by` lines, one
+/// `Reviewed-by` line, and one `Change-Id` line.
+pub fn get_footer_lines(body: &str) -> Option<IndexMap<String, Vec<String>>> {
```

Comment:
Can we use an empty map where you currently use `None`? I think it's simpler that way. Or does `None` mean something else than an empty map would mean?

Reply:
I think initially I was going to make this its own type with its own trait methods, but let the `IndexMap` leak through in this first draft.

I think if we make it a struct, it can probably also include writing the footers back out, to clean stuff out of `gerrit.rs`.